library/Fission/Internal/Orphanage/CID.hs

```diff
@@ -0,0 +1,8 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Fission.Internal.Orphanage.CID () where
+
+import Database.Persist.TH
+import Network.IPFS.CID.Types
+
+derivePersistField "CID"
```

Comment:
`derivePersistField` is a macro* that makes `CID` work directly in a DB table.

*Not technically a macro, but very similar idea